ui/explore/src/components/PanelEditor/PanelEditorForm.tsx

```diff
@@ -0,0 +1,112 @@
+import { ExploreToolbar } from '../ExploreToolbar';
+import { useTimeSeriesQueryEditorActions } from '../../context/useTimeSeriesQueryEditorActions';
+import { usePanelEditor } from '../../context/usePanelEditor';
+
+export interface PanelEditorProps {
+  initialValues: PanelEditorValues;
+  initialAction: Action;
+  exploreTitleComponent?: JSX.Element;
+}
+
+export interface PanelSpecEditorProps {
+  panelDefinition: PanelDefinition;
+  onQueriesChange: (queries: QueryDefinition[]) => void;
+  onPluginSpecChange: (spec: UnknownSpec) => void;
+}
+
+export function PanelEditorForm(props: PanelEditorProps) {
+  const {
+    exploreTitleComponent,
+    initialValues: { panelDefinition: initialPanelDef },
+  } = props;
+
+  const { setQueries, setPlugin, panelDefinition } = usePanelEditor(initialPanelDef);
+  const { plugin } = panelDefinition.spec;
+  const queries = panelDefinition.spec.queries ?? [];
+
+  // Use common plugin editor logic even though we've split the inputs up in this form
+  const pluginEditor = usePluginEditor({
+    pluginType: 'Panel',
+    value: { kind: plugin.kind, spec: plugin.spec },
+    onChange: (plugin: SetStateAction<Definition<UnknownSpec>>) => {
+      setPlugin(plugin);
+    },
+    onHideQueryEditorChange: (isHidden: boolean | undefined) => {
+      setQueries(undefined, isHidden);
+    },
+  });
+
+  const onChange = (queries: QueryDefinition[]) => setQueries(queries);
+
+  const {
+    handleQueryChange,
+    handleQueryAdd,
+    handleQueryCollapseExpand,
+    handleQueryDelete,
+    queriesCollapsed,
+    defaultTimeSeriesQueryKind,
+  } = useTimeSeriesQueryEditorActions({ onChange, queries });
+
+  const form = useForm({
+    mode: 'onBlur',
+    defaultValues: {
+      type: pluginEditor.pendingKind ? pluginEditor.pendingKind : plugin.kind,
+    },
+  });
+
+  return (
+    <>
+      <ExploreToolbar exploreTitleComponent={exploreTitleComponent} onQueryAdd={handleQueryAdd} />
+      <FormProvider {...form}>
+        <Grid container spacing={2}>
+          <Grid item xs={12}>
+            {/* Maybe in the future we should have something like PanelSpecEditor as we have on the Dashboard System */}
+            <TimeSeriesQueryEditor
+              handleQueryChange={handleQueryChange}
+              handleQueryCollapseExpand={handleQueryCollapseExpand}
+              handleQueryDelete={handleQueryDelete}
+              queriesCollapsed={queriesCollapsed}
+              queries={queries}
+              defaultTimeSeriesQueryKind={defaultTimeSeriesQueryKind}
+            />
+          </Grid>
+          <Grid item xs={12}>
+            <Box
+              sx={{
+                paddingX: 1,
+              }}
+            >
+              <Typography variant="h4" marginBottom={1}>
+                Preview
+              </Typography>
+              <ErrorBoundary FallbackComponent={ErrorAlert}>
+                <PanelPreview panelDefinition={panelDefinition} />
```

Comment:
I think using the TimeSeriesChart panel directly would be more composable that using PanelPreview, it’d also be neat to always show the table legend in this case since we know there is room when in explore? Here are some examples:

https://github.com/sjcobb/prometheus/blob/sjcobb/mui-refactor-prototype/web/ui/react-app/src/components/perses/TimeSeriesPanel.tsx


https://github.com/sjcobb/prometheus/blob/sjcobb/mui-refactor-prototype/web/ui/react-app/src/components/perses/PersesQueryPanel.tsx#L57

Reply:
Well I guess I’m too focused on time series queries in my comments actually, maybe the isExplore approach is okay after all 😅

Reply:
Sorry for the stream of consciousness but my next question is are there changes we could make to the query editing on dashboards that would make this easier? It seems what we want is a more responsive experience that looks good in the panel edit drawer and in full screen explorer (side note: we could also make the panel edit drawer larger) plus some way to hide or show that ‘Projects’ dropdown. 

I just don’t totally understand why the UX would be different enough to warrant the isExplore prop 🤔

Reply:
Hey @sjcobb nothing to apologize mate ❤️ 
We'd like to reuse the entire plugin system we have but with a different flavor in terms of UI for the explore page.
I thougth about have a dedicated TimeSeries Plugin for explore page but @Nexucis @jgbernalp and I we think it might be a bit complex maintain both plugins.

In theory I'm expecing the TimeSeries Plugin be less couple to the UI, but currently it isn't (that's totally Ok), I'm open for a better suggestion on how to implement this tbh, I didn't have any better idea.

Reply:
I have some ideas but I don’t want to hold this up, I really need to start by pulling the changes down and